rest/oidc_api.go

```diff
@@ -118,6 +149,26 @@ func (h *handler) handleOIDCCallback() error {
 		return base.HTTPErrorf(http.StatusBadRequest, "Unable to identify provider for callback request")
 	}
 
+	// Validate state parameter to prevent cross-site request forgery (CSRF) when callback state is enabled.
+	if !provider.DisableCallbackState {
+		stateCookie, err := h.rq.Cookie(stateCookieName)
+
+		if err != nil && err != http.ErrNoCookie {
+			base.Warnf("Unexpected error returned from request.Cookie(%s): %v", stateCookieName, err)
+			return ErrReadStateCookie
+		}
+		if err != nil || stateCookie == nil {
+			return ErrNoStateCookie
+		}
```

Comment:
The error checks here are not very readable - it takes a close reading to work out that line 161 is triggered when err == http.ErrNoCookie or stateCookie ==nil.  I think this would be clearer as:
```
		if err == http.ErrNoCookie || stateCookie == nil {
			return ErrNoStateCookie
		} 
		if err != nil  {
			base.Warnf("Unexpected error returned from request.Cookie(%s): %v", stateCookieName, err)
			return ErrReadStateCookie
		}
```